mlfinlab/tests/test_fingerpint.py

```diff
@@ -0,0 +1,125 @@
+
+        reg_1_fingerpint_70.fit()
+        reg_2_fingerpint_70.fit()
+
+        # Increasing the number of samples doesn't change feature effect massively
+        for feature in informative_features_1:
+            self.assertAlmostEqual(self.reg_1_fingerprint.linear_effect[feature],
+                                   reg_1_fingerpint_70.linear_effect[feature], delta=0.05)
+
+        for feature in informative_features_2:
+            self.assertAlmostEqual(self.reg_2_fingerprint.linear_effect[feature],
+                                   reg_2_fingerpint_70.linear_effect[feature], delta=0.05)
+
+    def test_non_linear_effect(self):
+        """
+        Test get_non_linear_effect for various regression models and num_values.
+        """
+
+        # Test the most informative feature effects for reg_1
+        informative_features_1 = [0, 5, 6, 12]
+        for feature, effect_value in zip(informative_features_1, [0.0758, 0.3848, 0.1, 0.28]):
+            self.assertAlmostEqual(self.reg_1_fingerprint.non_linear_effect[feature], effect_value, delta=1e-3)
+
+        # Non-linear effect to linear model is zero
+        for effect_value in self.reg_2_fingerprint.non_linear_effect.values():
+            self.assertAlmostEqual(effect_value, 0, delta=1e-8)
+
+        # Test fingerprints with bigger num_values
+        reg_1_fingerpint_70 = RegressionModelFingerprint(self.reg_1, self.X, num_values=70)
+        reg_2_fingerpint_70 = RegressionModelFingerprint(self.reg_2, self.X, num_values=70)
+
+        reg_1_fingerpint_70.fit()
+        reg_2_fingerpint_70.fit()
+
+        # Increasing the number of samples doesn't change feature effect massively
+        for feature in informative_features_1:
+            self.assertAlmostEqual(self.reg_1_fingerprint.non_linear_effect[feature],
+                                   reg_1_fingerpint_70.non_linear_effect[feature], delta=0.05)
+
+        for effect_value in reg_2_fingerpint_70.non_linear_effect.values():
+            self.assertAlmostEqual(effect_value, 0, delta=1e-8)
+
+    def test_pairwise_effect(self):
+        """
+        Test get_pairwise_effect for various regression models and num_values.
+        """
+
+        combinations = [(0, 5), (0, 12), (1, 2), (5, 7), (3, 6), (4, 9)]
+        self.reg_1_fingerprint.get_pairwise_effect(combinations)
+        self.reg_2_fingerprint.get_pairwise_effect(combinations)
+
+        for pair, effect_value in zip(combinations, [0.167, 0.167, 0.166, 0.166, 0.166, 0.166]):
+            self.assertAlmostEqual(self.reg_1_fingerprint.pair_wise_effect[str(pair)], effect_value, delta=1e-3)
+
+        # Pairwise effect for linear model should be the same for all combinations
+        for pair in combinations:
+            self.assertAlmostEqual(self.reg_2_fingerprint.pair_wise_effect[str((0, 5))],
+                                   self.reg_2_fingerprint.pair_wise_effect[str(pair)], delta=1e-9)
+
+    def test_plot_effects(self):
```

Comment:
This function doesn't have any assertions?

Does it test anything?

Reply:
As it just makes a plot, I don't know what to test for in terms of assertions

Reply:
Hmm, is there a way we can just exclude the plot_effects code from coverage?

Reply:
As far as I understand, we need to reconfigure travis for that somehow

Reply:
Ok leave this unit test here for now. I will see if I can figure it out.